bosswave.cpp

```diff
@@ -7,6 +7,50 @@
 
 #include <msgpack.h>
 
+#include <cstdio>
+
+class NotImplementedException : public std::exception
```

Comment:
Qt isn't exception-safe, but it usually works (see http://doc.qt.io/qt-5/exceptionsafety.html).

I only throw an exception in code paths indicative of a bug somewhere in the bindings or the router. In my opinion, it doesn't need to be documented, since the user should never see them under normal circumstances.